src/config.ts

```diff
@@ -1,22 +1,31 @@
-// add a .env file to your project to set these environment variables
+// add a ".env" file to your project to set these environment variables
+import { INTENT } from "@paypal/paypal-js";
 import * as dotenv from "dotenv";
 dotenv.config();
 
 const {
-  PAYPAL_CLIENT_ID,
-  PAYPAL_CLIENT_SECRET,
+  PAYPAL_ENVIRONMENT_MODE,
+  PAYPAL_SANDBOX_CLIENT_ID,
+  PAYPAL_SANDBOX_CLIENT_SECRET,
+  PAYPAL_LIVE_CLIENT_ID,
+  PAYPAL_LIVE_CLIENT_SECRET,
+  PAYPAL_CURRENCY,
+  PAYPAL_INTENT,
   PAYPAL_API_BASE_URL,
   PAYPAL_WEB_BASE_URL,
 } = process.env;
 
 function getConfig() {
+  const env = PAYPAL_ENVIRONMENT_MODE?.toLowerCase() || "sandbox";
   return {
     paypal: {
-      clientID: PAYPAL_CLIENT_ID,
-      clientSecret: PAYPAL_CLIENT_SECRET,
-      apiBaseUrl: PAYPAL_API_BASE_URL || "https://api-m.sandbox.paypal.com",
+      clientID: env === "sandbox" ? PAYPAL_SANDBOX_CLIENT_ID : PAYPAL_LIVE_CLIENT_ID,
+      clientSecret: env === "sandbox" ? PAYPAL_SANDBOX_CLIENT_SECRET : PAYPAL_LIVE_CLIENT_SECRET,
+      currency: PAYPAL_CURRENCY || "USD",
+      intent: PAYPAL_INTENT?.toUpperCase() as INTENT || "CAPTURE" as INTENT,
```

Comment:
I am a little hesitant to have currency and intent in the config. It does assume that these values are set once at server start up and do not change. 

For example, we do have large merchants that show the PayPal button in two different ways depending on which page its on (shopping cart page vs final checkout page).

I haven't seen how global merchants configure currency but do wonder if they have separate web sites for each country (where we can set up currency at server start up time) or if this value would be dynamic based a run time setting 🤔 

I want to give this one more thought.

Reply:
The 99.5% use case is to have a single currency and intent. Some may want it to be configurable in a control panel setting instead of the environment, and if so they'll need to add that functionality. For those that need multiple currencies/intents, they'll have to add logic where and when they want it to load the SDK dynamically and have the API calls be in sync with it. Certainly doable, but I think out of scope for this example. Overall for the 99.5% users, for those of them who do need to set a different currency or intent I think having a simple way to keep the SDK in sync with the order creation is worth including 🙂